utilities/cli/holohub.py

```diff
@@ -1466,6 +1416,72 @@ def handle_setup(self, args: argparse.Namespace) -> None:
 
+    def _setup_cuda_packages(self, cuda_major_version: int, dryrun: bool = False) -> None:
+        """Find and install CUDA packages for Holoscan SDK development"""
+
+        # Attempt to install cudnn9
+        # cuDNN version example: libcudnn9-cuda-12/unknown,now 9.10.2.21-1 amd64
+        CUDNN_9_PATTERN = r"9\.[0-9]+\.[0-9]+\.[0-9]+\-[0-9]+"
+        installed_cudnn9_version = holohub_cli_util.install_cuda_dependencies_package(
+            package_name="libcudnn9-cuda-12",
+            preferred_version=CUDNN_9_PATTERN,
+            optional=True,
+            dry_run=dryrun,
+        )
+        holohub_cli_util.install_cuda_dependencies_package(
+            package_name="libcudnn9-dev-cuda-12",
+            preferred_version=re.escape(installed_cudnn9_version),
+            optional=True,
+            dry_run=dryrun,
+        )
+
+        # Validate cudnn9 installation and fall back to cudnn8 if necessary
+        try:
+            if not installed_cudnn9_version:
+                # cuDNN version example: libcudnn8/unknown 8.9.7.29-1+cuda12.2 amd64
+                CUDNN_8_PATTERN = (
+                    rf"8\.[0-9]+\.[0-9]+\.[0-9]+\-[0-9]\+cuda{cuda_major_version}\.[0-9]+"
+                )
+                installed_cudnn8_version = holohub_cli_util.install_cuda_dependencies_package(
+                    package_name="libcudnn8",
+                    preferred_version=CUDNN_8_PATTERN,
+                    dry_run=dryrun,
+                )
+                holohub_cli_util.install_cuda_dependencies_package(
+                    package_name="libcudnn8-dev",
+                    preferred_version=re.escape(installed_cudnn8_version),
+                    dry_run=dryrun,
+                )
+        except subprocess.CalledProcessError as e:
+            print(f"Error checking cudnn9 version, skipping: {e}")
+
+        # Install TensorRT dependencies
+        # TensorRT version example: libnvinfer-bin/unknown,now 10.12.0.36-1+cuda12.9 amd64
+        NVINFER_PATTERN = rf"\d+\.[0-9]+\.[0-9]+\.[0-9]+-[0-9]\+cuda{cuda_major_version}\.[0-9]+"
+        installed_libnvinferbin_version = holohub_cli_util.install_cuda_dependencies_package(
+            package_name="libnvinfer-bin",
+            preferred_version=NVINFER_PATTERN,
+            optional=True,
+            dry_run=dryrun,
+        )
+        if not installed_libnvinferbin_version:
+            holohub_cli_util.fatal(
+                f"Failed to install libnvinfer-bin for CUDA {cuda_major_version}"
+            )
+        libnvinfer_pattern = re.escape(installed_libnvinferbin_version)
+
+        for trt_package_name in [
+            "libnvinfer-headers-dev",
+            "libnvinfer-dev",
+            "libnvinfer-plugin-dev",
+            "libnvonnxparsers-dev",
+        ]:
+            holohub_cli_util.install_cuda_dependencies_package(
+                package_name=trt_package_name,
+                preferred_version=libnvinfer_pattern,
```

Comment:
I think this should be set to optional=True to install the packages as much as possible?

Reply:
No strong guidance for behavior when any individual package in a set fails installation, so long as behavior is well defined. I've updated to fail out as soon as possible on failure to save time on an incomplete installation, open to feedback.